### PR TITLE
test: expose test-jar and mock classes in oauth2

### PR DESCRIPTION
**Pull request by diegomarquezp**
We need to expose some test utils to be used in https://github.com/googleapis/sdk-platform-java/pull/1642

Basically to allow `GdchCredentials` to be refreshed via fake transport

### CODE REVIEW

File: appengine/pom.xml [hunk 1]
```diff
@@ -75,8 +75,10 @@
     <dependency>
       <groupId>com.google.auth</groupId>
       <artifactId>google-auth-library-oauth2-http</artifactId>
-      <type>test-jar</type>
+      <version>${project.version}</version>
```
Comment (blakeli0) [inline appengine/pom.xml:78]: Do we need the version? I think it is already configured in the parent pom?

File: oauth2_http/javatests/com/google/auth/oauth2/MockHttpTransportFactory.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package com.google.auth.oauth2;
+
+import com.google.api.client.http.HttpTransport;
+import com.google.api.client.testing.http.MockHttpTransport;
+import com.google.auth.http.HttpTransportFactory;
+
+class MockHttpTransportFactory implements HttpTransportFactory {
```
Comment (blakeli0) [inline oauth2_http/javatests/com/google/auth/oauth2/MockHttpTransportFactory.java:7]: If you plan to use it in other repos, this needs to be a public class.

**Review by blakeli0 — APPROVED** [review]

**Review by blakeli0 — CHANGES_REQUESTED** [review]
Didn't mean to approve the PR, please resolve the comments before merging.

File: appengine/pom.xml (hunk 1 above)
Comment (diegomarquezp) [reply appengine/pom.xml]: I removed it, thanks for the catch.

File: oauth2_http/javatests/com/google/auth/oauth2/MockHttpTransportFactory.java (hunk 2 above)
Comment (diegomarquezp) [reply oauth2_http/javatests/com/google/auth/oauth2/MockHttpTransportFactory.java]: Seems like it was the only exposed class that needed a modifier. Changed

**Review by TimurSadykov — APPROVED** [review]
LGTM

**Review by blakeli0 — APPROVED** [review]

**Pull request merged**